### PR TITLE
Add tests for TagsAggregator.

### PR DESCRIPTION
**Pull request by sheetaluk**

### CODE REVIEW

File: tests/h/search/query_test.py [hunk 1]
```diff
@@ -505,11 +505,32 @@ def test_matches_replies_of_replies_to_an_annotation(self, Annotation, search):
...
+
+    def test_it_limits_annotation_aggregation_on_tags(self, Annotation, search):
+        ann1 = Annotation(tags=['tag_a'])
+        ann2 = Annotation(tags=['tag_a'])
+        ann3 = Annotation(tags=['tag_a'])
+
+        search.append_aggregation(query.TagsAggregation(limit=2))
+        result = search.run({})
+
+        count_for_tag_a = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_a")['count']
```
Comment (sheetaluk) [inline tests/h/search/query_test.py]: Shouldn't this count returned be limited to 2?

File: tests/h/search/query_test.py [hunk 2]
```diff
@@ -505,11 +505,32 @@ def test_matches_replies_of_replies_to_an_annotation(self, Annotation, search):
...
+        for i in range(3):
+            Annotation(tags=['tag_a'])
+
+        search.append_aggregation(query.TagsAggregation(2))
+        result = search.run({})
+
+        count_for_tag_a = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_a")['count']
+
+        assert len(result.aggregations.get('tags')) is 1
+        assert count_for_tag_a == 3
```
Comment (sheetaluk) [inline tests/h/search/query_test.py]: Shouldn't this count returned be limited to 2? It should not be 3 if I've changed the limit to be 2.

File: tests/h/search/query_test.py [hunk 3]
```diff
@@ -505,11 +505,32 @@ def test_matches_replies_of_replies_to_an_annotation(self, Annotation, search):
...
+        for i in range(3):
+            Annotation(tags=['tag_a'])
+
+        search.append_aggregation(query.TagsAggregation(2))
+        result = search.run({})
+
+        count_for_tag_a = next(r for r in result.aggregations.get('tags') if r["tag"] == "tag_a")['count']
+
+        assert len(result.aggregations.get('tags')) is 1
+        assert count_for_tag_a is 3
```
Comment (sheetaluk) [inline tests/h/search/query_test.py]: Shouldn't this count returned be limited to 2? It should not be 3 if I've changed the limit to be 2.

Comment (robertknight) [reply tests/h/search/query_test.py]: See https://github.com/hypothesis/h/pull/5130#discussion_r200955163

**Pull request merged**